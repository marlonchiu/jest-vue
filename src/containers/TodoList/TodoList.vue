<template>
  <div class="todo-list-Page">
    <Header @add="addUndoItem" />
    <UndoList
      :list="undoList"
      @delete="handleItemDelete"
      @status="changeItemStatus"
      @reset="resetItemStatus"
      @change="changeItemValue"
    />
  </div>
</template>

<script>
import axios from 'axios'
import Header from './components/Header'
import UndoList from './components/UndoList'
export default {
  name: 'TodoList',
  components: {
    Header,
    UndoList
  },
  data () {
    return {
      undoList: []
    }
  },
  mounted () {
    axios.get('/getUndoList.json').then((res) => {
      // console.log(res.data)
      this.undoList = res.data
    }).catch((e) => {
      console.log(e)
    })
  },
  methods: {
    addUndoItem (data) {
      this.undoList.push({
        value: data,
        status: 'div'
      })
    },
    handleItemDelete (index) {
      this.undoList.splice(index, 1)
    },
    changeItemStatus (index) {
      const newList = []
      this.undoList.forEach((item, itemIndex) => {
        if (itemIndex === index) {
          newList.push({
            value: item.value,
            status: 'input'
          })
        } else {
          newList.push(item)
        }
      })

      this.undoList = newList
    },
    resetItemStatus () {
      const newList = []
      this.undoList.forEach(({ value }) => {
        newList.push({
          value,
          status: 'div'
        })
      })
      this.undoList = newList
    },
    changeItemValue (newItem) {
      const { value, index } = newItem
      this.undoList[index].value = value
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="stylus">
</style>
