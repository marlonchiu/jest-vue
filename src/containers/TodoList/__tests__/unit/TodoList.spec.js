import { shallowMount } from '@vue/test-utils'
import TodoList from '../../TodoList.vue'
import Header from '../../components/Header'

describe('TodoList.vue', () => {
  it ('组件渲染正常', () => {
    const wrapper = shallowMount(TodoList)
  })

  it ('TodoList 初始化的时候， undoList 为空', () => {
    const wrapper = shallowMount(TodoList)
    const undoList = wrapper.vm.$data.undoList
    expect(undoList).toEqual([])
  })

  it('TodoList 监听到 Header 的 add 事件的时候， undoList 增加一个内容', () => {
    const content = 'dell lee'
    const wrapper = shallowMount(TodoList)
    const header = wrapper.find(Header)
    header.vm.$emit('add', content)
    const undoList = wrapper.vm.$data.undoList
    expect(undoList).toEqual([content])
  })
})
