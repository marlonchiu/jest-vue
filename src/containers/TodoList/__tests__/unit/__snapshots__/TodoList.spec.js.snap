// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList.vue 样式改变 做提示 1`] = `
<div class="todo-list-Page">
  <header-stub></header-stub>
  <undolist-stub list=""></undolist-stub>
</div>
`;
